manager/orchestrator/task.go

```diff
@@ -6,9 +6,19 @@ import (
 
 	"github.com/docker/swarmkit/api"
 	"github.com/docker/swarmkit/identity"
+	"github.com/docker/swarmkit/log"
+	"github.com/docker/swarmkit/manager/state/store"
 	"github.com/docker/swarmkit/protobuf/ptypes"
+	gogotypes "github.com/gogo/protobuf/types"
+	"golang.org/x/net/context"
 )
 
+// Restarter defines an interface to start or delay start tasks
+type restarter interface {
+	StartNow(tx store.Tx, taskID string) error
+	DelayStart(ctx context.Context, _ store.Tx, oldTask *api.Task, newTaskID string, delay time.Duration, waitStop bool) <-chan struct{}
+}
```

Comment:
As an alternative to this interface, we could introduce an `orchestrator/init` package that can import both `orchestrator` and `orchestrator/restart`.